.github/workflows/build-kubespray-image.yaml

```diff
@@ -0,0 +1,47 @@
+---
+name: Build kubespray images
+
+on:
+  push:
+    branches: [master, release-.*]
+  release:
+    branches: [master, release-.*]
+
+env:
+  IMAGE_REPO: "kubespray"
+  IMAGE_NAME: "kubespray"
+  IMAGE_REGISTRY: "hub.k8s.li"
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Define image tags by current git repo tag and branch
+        shell: bash
+        run: |
+          git describe --tags --always | sed 's/^/IMAGE_TAG=/' >> $GITHUB_ENV
+          git branch --show-current | sed 's/master/latest/;s/release-//;s/^/IMAGE_TAG_BY_BRANCH=/' >> $GITHUB_ENV
+
+      - name: Build image
+        shell: bash
+        run: |
+          docker build -t ${IMAGE_REGISTRY}/${IMAGE_REPO}/${IMAGE_NAME}:${IMAGE_TAG} -f Dockerfile .
+
+      - name: Login to image registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.IMAGE_REGISTRY }}
+          username: ${{ secrets.REGISTRY_USERNAME }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+
+      - name: Push images to registry
+        shell: bash
+        run: |
+          docker tag ${IMAGE_REGISTRY}/${IMAGE_REPO}/${IMAGE_NAME}:${IMAGE_TAG} ${IMAGE_REGISTRY}/${IMAGE_REPO}/${IMAGE_NAME}:${IMAGE_TAG_BY_BRANCH}
+          docker push ${IMAGE_REGISTRY}/${IMAGE_REPO}/${IMAGE_NAME}:${IMAGE_TAG}
+          docker push ${IMAGE_REGISTRY}/${IMAGE_REPO}/${IMAGE_NAME}:${IMAGE_TAG_BY_BRANCH}
```

Comment:
Just a question: Why do we need to push 2 different tags with the same image?

Reply:
`IMAGE_TAG ` is defined by git tag,  it's a specific commit such as `v2.16.0` and `v2.15.0-252-gb1b407a0`. The main purpose here is to automatically build released version like `v2.16.0`.

`IMAGE_TAG_BY_BRANCH` is defined by git branch, it's the latest commit of the current branch. The main purpose here is to automatically build the latest image of the current branch like `release-2.16`.

Reply:
For example: `ubuntu:bionic-20200807` is a specific released version, and `ubuntu:bionic` is the latest version. On this date 20200807, `ubuntu:bionic-20200807` and `ubuntu:bionic` are identical.

For kubespray, when release a new version like v2.16.0, this job will push `kubespray:v2.16.0` by tag and `kubespray:v2.16` by release-2.16 branch. When a new PR is merged into release-2.16 branch, this job will push `kubespray:v2.15.0-1-gb1b407a0` by tag and the latest `kubespray:v2.16` by branch.

Reply:
Thanks for your explanation, I got it.